test/e2e/unmanaged_dns_test.go

```diff
@@ -0,0 +1,287 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	iov1 "github.com/openshift/api/operatoringress/v1"
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+	ingresscontroller "github.com/openshift/cluster-ingress-operator/pkg/operator/controller/ingress"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestUnmanagedDNSToManagedIngressController(t *testing.T) {
```

Comment:
*nit* if you look at the names of the e2e tests, they are not quite consistent:
`TestUnmanagedDNSToManagedIngressController`
`TestManagedDNSToUnmanagedDNSIngressController`
`TestUnmanagedDNSToManagedInternalIngressController`
Maybe just put DNS always after Unmanaged/Managed.